### ISSUE TITLE
mapping and rfc7946 ring ordering

### ISSUE THREAD

**Issue by rbuffat** [shapely/shapely#1319]
[RFC7946](https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.6) specifies an ordering for rings:

> A linear ring MUST follow the right-hand rule with respect to the
>       area it bounds, i.e., exterior rings are counterclockwise, and
>       holes are clockwise.
> 
>    Note: the [[GJ2008](https://datatracker.ietf.org/doc/html/rfc7946#ref-GJ2008)] specification did not discuss linear ring winding
>    order.  For backwards compatibility, parsers SHOULD NOT reject
>    Polygons that do not follow the right-hand rule.

Currently, mapping does not change / correct the ordering of rings. Should there be an option in mapping to generate GeoJSON Polygons according to RFC7946?

E.g. by orienting the rings first using something like 

```
    if isinstance(geometry, Polygon):
        return orient(geometry, sign=1.0) # type: ignore
    elif isinstance(geometry, MultiPolygon): # type: ignore
        return MultiPolygon([orient(geom, sign=1.0) for geom in geometry.geoms]) # type: ignore
```

**Comment by jorisvandenbossche** [comment]
That sounds as a useful option, yes. 

I was looking at GEOS if they actually also support this, but it's not exposed in the C API. It is a bit unfortunate that the `normalize()` method (`GEOSNormalize`) seems to do the exact opposite than what we want here (outer ring clockwise, inner rings counter clockwise)

**Comment by rkaravia** [comment]
With shapely 2.1.0 and [`shapely.orient_polygons()`](https://shapely.readthedocs.io/en/2.1.0/reference/shapely.orient_polygons.html) this looks relatively straightforward, see prototype in #2297, which adds a `force_ccw` flag to `mapping()`.

I would be happy to put in some more work to get this to a mergeable state (such as adding more tests/docs), but I'm not sure if the whole idea really makes sense.

From the docs about [mapping](https://shapely.readthedocs.io/en/2.1.0/manual.html#shapely.geometry.mapping), I saw that `mapping` should also work for "any object which implements `__geo_interface__`". This would require conversion to a shapely geometry before the call to `orient_polygons()`. I think it would be a bit weird to convert to a shapely geometry conditionally based on the `force_ccw` flag. What do you think?

**Comment by rkaravia** [comment]
Thank you for the feedback! 

Yes, as I wrote above, I also have some doubts about this.

I'm currently using the method you propose to get RFC 7946 GeoJSON, but it took me some digging through the docs to figure it out.

Would it make sense to mention something about polygon orientation in the `mapping()` and `to_geojson()` docs?

**Comment by sgillies** [comment]
@rkaravia yes, I think it would be good to have a mention about orientation and use of `orient_polygons()` in the docs of either or both of those functions.
